File: platform/windows/AdGuard.FilterListManager/MarshalLogic/FilterGroup.cs
namespace AdGuard.FilterListManager.MarshalLogic
{
    public class FilterGroup
    {
        public FilterGroup(int id, string name, int displayNumber)
        {
            this.id = id;
            this.name = name;
            this.displayNumber = displayNumber;
        }

        public int id { get; set; }
        public string name { get; set; }
        public int displayNumber { get; set; }

        public void Deconstruct(out int id, out string name, out int displayNumber)
        {
            id = this.id;
            name = this.name;
            displayNumber = this.displayNumber;
        }
    }
}

File: crates/filter-list-manager/src/manager/models/filter_group.rs
//! FilterListGroup represents a group of filter lists that have similar purpose.
use serde::{Deserialize, Serialize};

/// FilterListGroup represents a group of filter lists that have similar
/// purpose.
#[derive(Clone, Serialize, Deserialize, Debug)]
pub struct FilterGroup {
    /// Group id
    pub id: i32,
    /// Group name
    pub name: String,
    /// Display number for ordering
    pub display_number: i32,
}
